### PR TITLE
Backport z-level drawing, the second take

### PR DESCRIPTION
**Pull request by sonphantrung**
#### Summary
Features "z-level drawing backported to BN"

#### Purpose of change
I want to see zeds from below, 2nd attempt of #61074

#### Describe the solution
Backport the change from [BN](https://github.com/cataclysmbnteam/Cataclysm-BN/pull/1743) by @joveeater, with a slightly few changes (the deprecated usage of `g->` has been replaced)

#### Describe alternatives you've considered
Not do this

#### Testing
TODO:
- [ ] Compile this successfully
- [ ] Deal with problems inherited from the fact that BN's codebase is OLD
- [ ] (maybe?) have ncurses support

#### Additional context
I am not a C++ guy

### CODE REVIEW

**Comment by Fris0uman** [comment]
> I am not a C++ guy
I am a bit confused at how you plan to manage to get this to work without knowing how to actually code the feature

**Comment by sonphantrung** [comment]
I mean, I know how to deal with editing, tweaking the current code, but I'm fairly dumb in actually further extending the functionality of that same code

**Comment by EternalKitten** [comment]
I believe in you, little shark.

File: src/cata_tiles.cpp [hunk 1]
```diff
@@ -1630,7 +1630,7 @@ void cata_tiles::draw( const point &dest, const tripoint &center, int width, int
     };
 
     const std::array < zlevel_layer, 3 > zlevel_drawing_layers = {{
-            {true, &cata_tiles::draw_field_or_item}, {false, &cata_tiles::draw_vpart}, {true, &cata_tiles::draw_critter_at}
+            {true, &cata_tiles::draw_field_or_item}, {false, &cata_tiles::draw_vpart_roof}, {false, &cata_tiles::draw_vpart_no_roof}, {true, &cata_tiles::draw_critter_at}
```
Comment (mlange-42) [inline src/cata_tiles.cpp:1633]: Looks like this is the wrong drawing order. Did you check that in-game?

**Comment by sonphantrung** [comment]
This [line](https://github.com/CleverRaven/Cataclysm-DDA/pull/61365/files#diff-c8561880131eb6314d85f58032f9063284bcfc226b75b3b5df585dcaa219605cR1687) do bother me a lot

**Comment by Calavera20012** [comment]
Hello, I am interested in tackling this problem, have you made any progress or noted any important problems?

**Comment by sonphantrung** [comment]
Thanks! So far, I'm currently dealing with misc errors at compilation, which I don't have enough time to deal with for now.

**Comment by Fris0uman** [comment]
Closing in favor of #65738

**Pull request closed without merging**